crates/polars-utils/src/pl_str.rs

```diff
@@ -0,0 +1,124 @@
+use compact_str::CompactString;
+
+#[macro_export]
+macro_rules! format_pl_smallstr {
+    ($($arg:tt)*) => {{
+        use std::fmt::Write;
+
+        let mut string = PlSmallStr::const_default();
+        write!(string, $($arg)*).unwrap();
+        string
+    }}
+}
+
+/// String type that interns small strings and has O(1) clone.
+#[derive(Clone, Debug, PartialEq, Eq, Hash, PartialOrd, Ord)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct PlSmallStr(CompactString);
+
+impl PlSmallStr {
+    /// Initialize an empty string ""
+    #[inline(always)]
+    pub const fn const_default() -> Self {
+        Self(CompactString::const_new(""))
+    }
+
+    #[inline(always)]
+    pub const fn new_from_static(s: &'static str) -> Self {
+        Self(CompactString::const_new(s))
+    }
+
+    #[inline(always)]
+    pub fn new_from_str(s: &str) -> Self {
```

Comment:
The convention for explicit `from` constructors is that they don't also specify `new`. So instead of `new_from_static`, `new_from_str`, `new_from_string`, it should just be `from_static`, `from_str`, `from_string`.